test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -104,5 +116,21 @@ function ($, Q, Expect, $sandbox, DeviceFingerprint) {
       });
     });
 
+    it('ignores if message is not from right iframe', function(done) {
+      mockIFrameMessages(true);
+      DeviceFingerprint.generateDeviceFingerprint(baseUrl, $sandbox)
+        .then(function () {
+          done.fail('Fingerprint promise should not have been resolved');
+        })
+        .fail(function () {
+          done.fail('Fingerprint promise should not have been rejected');
+        });
+      Util.mockSetTimeout();
+      setTimeout(function() {
+        // give it time to check if promise resolves or rejects.
+        done();
+      }, 1000);
```

Comment:
does this mean wait for 1 second before succeeding? Is that a guaranteed time for the promise to return? Seems strange to have a wait in a unit test

Reply:
no we dont actually wait 1sec. We fake waiting for 1 sec using the mock Util.